pkg/chunk/chunk_store.go

```diff
@@ -150,18 +151,35 @@ func (c *Store) calculateDynamoWrites(userID string, chunks []Chunk) (WriteBatch
 }
 
 // Get implements ChunkStore
-func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]Chunk, error) {
-	logger := util.WithContext(ctx)
+func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]local.SeriesIterator, error) {
 	if through < from {
 		return nil, fmt.Errorf("invalid query, through < from (%d < %d)", through, from)
 	}
 
-	filters, matchers := util.SplitFiltersAndMatchers(allMatchers)
+	// Fetch metric name chunks if the matcher is of type equal,
+	metricNameMatcher, matchers, ok := util.ExtractMetricNameMatcherFromMatchers(allMatchers)
+	if ok && metricNameMatcher.Type == metric.Equal {
+		return c.getMetricNameIterators(ctx, from, through, matchers, metricNameMatcher.Value)
+	}
+
+	// Otherwise we will create lazy iterators for all series in our index
+	return c.getSeriesIterators(ctx, from, through, matchers, metricNameMatcher)
+}
 
-	// Fetch chunk descriptors (just ID really) from storage
-	chunks, err := c.lookupChunksByMatchers(ctx, from, through, matchers)
+func (c *Store) getMetricNameIterators(ctx context.Context, from, through model.Time, allMatchers []*metric.LabelMatcher, metricName model.LabelValue) ([]local.SeriesIterator, error) {
+	chunks, err := c.getMetricNameChunks(ctx, from, through, allMatchers, metricName)
```

Comment:
My preference would be to move the contents of `getMetricNameChunks` into here, as this function seems overly short.  I see that it is used from the tests, so you'd need to do some work there, in which case it might not be worth the effort.